ndindex/slice.py

```diff
@@ -338,3 +338,55 @@ def reduce(self, shape=None, axis=0):
             if -size <= stop < 0:
                 stop += size
         return self.__class__(start, stop, step)
+
+    def newshape(self, shape=None, axis=0):
+        """
+        `Slice.newshape(shape)` returns the shape of `a[idx.raw]`, assuming `a`
+        has shape `shape`.
+
+        `shape` can be an int, a tuple of ints, an Integer or a tuple of ints
+        or Integers.
+
+        A ValueError is raised if `shape` is `None`.
+
+        >>> from ndindex import Slice, Integer, Tuple
+        >>> idx = Slice(6)
+        >>> idx.newshape((8, 2))
+        (6, 2)
+        >>> idx.newshape(10)
+        (6,)
+        >>> idx.newshape((8, 10), axis=1)
+        (8, 6)
+        """
+        from . import Tuple
```

Comment:
Should be this at the top?

Reply:
You can try it but it might fail due to circular imports.